src/backend/tools/brave_search/tool.py

```diff
@@ -57,15 +57,20 @@ async def call(
                 AgentToolMetadataArtifactsType.DOMAIN, session, ctx
             )
 
-        response = await self.client.search_async(
-            q=query, count=self.num_results, include_domains=filtered_domains
-        )
+        try:
+            response = await self.client.search_async(
+                q=query, count=self.num_results, include_domains=filtered_domains
+            )
+        except Exception as e:
+            return self.get_tool_error(
+                ToolError(text=f"Error calling tool {self.ID}", details=str(e))
```

Comment:
I think instead of having a ToolError everywhere, could we instead pass in the `details` only to `self.get_tool_error()?`

Then `get_tool_error()` would instantiate a `ToolError` with `cls.ID`

Reply:
resolved